### DISCUSSION TITLE
How we can to apply filter on TextBox or IText?

### DISCUSSION THREAD

**Discussion by vinayITT** [fabricjs/fabric.js#9582] [Q&A]
<!DOCTYPE html>
<html>
  <head>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/fabric.js/5.3.0/fabric.min.js"></script>
  </head>

  <body>
    <canvas id="canvas"></canvas>
    <button onclick="createFilterFilter()">Apply Filter</button>

    <script>
      var canvas = new fabric.Canvas("canvas");
      canvas.setWidth(900);
      canvas.setHeight(500);
      canvas.set({ backgroundColor: "yellow" });

      function applyFilterToElement(element, filterName) {
        element.filters = [];
        if (filterName === "grayscale") {
          element.filters.push(new fabric.Image.filters.Grayscale());
        } else if (filterName === "brightness") {
          element.filters.push(
            new fabric.Image.filters.Brightness({ brightness: 0.5 })
          );
        }
        element.applyFilters();
        canvas.renderAll();
      }

      function createFilterFilter() {
        applyFilterToElement(textElement, "brightness");
      }

      const textElement = new fabric.IText(
        "QUICK NOTE \n HELLO, \n Filter ON TEXT",
        {
          fontFamily: "Sans-Serif",
          fontWeight: "bold",
          left: 90,
          top: 200,
          fontSize: 64,
          width: 450,
          strokeWidth: 1,
          stroke: "#000000",
          lineHeight: 0.7,
          type: "image",
        }
      );

      // Text modification
      textElement.on("modified", function () {});

      canvas.add(textElement);
      canvas.renderAll();
    </script>
  </body>
</html>

**Comment by ShaMan123** [comment]
It is in the roadmap
For now you can workaround it 2 ways:

1. Overriding the `render` method and/or `renderCache`
3. Use an image `new FabricImage(textbox.toCanvasElement()).applyFilters()`

**Reply by vinayITT** [reply]
@ShaMan123 Thanks for your response.
As you suggested, the second one looks like a lot in a case where we have to manage editing of text on canvas. Because once the text rendered as a image we can not edit it. 
 I am bit curious about the first one, It will be very helpful if you can share some code reference for the first one.
**Overriding the render method and/or renderCache**

**Reply by ShaMan123** [reply]
Both approaches are the same in terms of rendering
You render the text onto one context, apply a filter to the context and render it to the main canvas context.
You can use the applyFilters logic for that I believe
